File: search/binary_search.py
arr = [_ for _ in range(100)]


def binary_search(_list, start, end, value):
    mid = (end + start) // 2
    if end - start == 1:
        if _list[start] != value:
            return None
        else:
            return start
    if _list[mid] == value:
        return mid
    if value < _list[mid] and mid > start:
        return binary_search(_list, start, mid, value)
    elif value > _list[mid] and mid < end:
        return binary_search(_list, mid, end, value)


if __name__ == '__main__':
    index = binary_search(arr, 0, len(arr), 25)
    print(index)

File: search_js/binary_search.js
var arr = [1, 4, 7, 9, 23, 24, 33, 41, 49, 66, 78, 92, 99, 100];

function binary_search(a, start, end, val) {
    if (end - start === 1) {
        var res;
        a[start] === val ? res = start : res = undefined;
        return res;
    }
    var mid = parseInt((end + start) / 2);
    if (a[mid] > val) {
        return binary_search(a, start, mid, val);
    }
    else if (a[mid] < val) {
        return binary_search(a, mid, end, val);
    }
    else
        return mid;
}

(function main() {
    var target = 33;
    var index = binary_search(arr, 0, arr.length, target);
    console.log(index);
})();